packages/stream-json/reader.js

```diff
@@ -0,0 +1,26 @@
+// @ts-check
```

Comment:
Huh. I just noticed. Are we not following the `src` vs `test` directory convention? I think I like that better than having a bunch of source files at the package level, but I have not thought about it.

Reply:
The rule is generally the same, but I am proposing an amendment and I think I have @michaelfig’s blessing:

With Node.js implementing ESM they added `imports` and `exports` features to `package.json` that make it possible to expose more than `main` as the package’s exported API surface, and this is now *enforced*. But, to make a package that straddles the old world and the new, you have to keep the `.js` extension.

It has always been a faux pas to reach like `import "@org/package/src/internal.js`. The `src` directory should indicate that the modules within are not participating in the public API. It’s more proper to `import '@netstring/reader.js'`. It is possible with an `exports` directive to even make this `import '@netstring/reader'`, just not portably.

So I’m proposing that we surface all public API modules at the top-level. It would certainly be consistent with our existing rule to require each of these a thunk for an internal module, e.g., `export * from './src/internal.js'`, but I feel that would be wasteful and fragile.